### PR TITLE
feat/javadoc: requirement for automating the generation of javadoc

### PR DESCRIPTION
**Pull request by samirromdhani**
requirement for automating the generation of javadoc based gh-pages

### CODE REVIEW

File: docs/examples/example-app/pom.xml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      <url>https://maven.pkg.github.com/com-pas/*</url>
+    </repository>
+  </repositories>
+
+  <dependencies>
+    <!-- SclService -->
+    <dependency>
+      <groupId>org.lfenergy.compas</groupId>
+      <artifactId>sct-commons</artifactId>
+      <version>0.1.0</version>
```
[the remark this answers is not in the document]
Comment (samirromdhani) [reply docs/examples/example-app/pom.xml]: done

File: docs/QUICK_START.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+**sct-commons** provides a collection of services to help you build SCL files for a range of use cases.
+Following services provides needed functions compliant with IEC 61850 :
+
+1. **SclService**
+2. **SubstationService**
+
+Let's start with a simple **SclService** call
+
+```java
+var scl = SclService.initScl(Optional.empty(), "1.0", "1.0");
```
[the remark this answers is not in the document]
Comment (samirromdhani) [reply docs/QUICK_START.md:48]: You can explain.
I just reported these in `gh-pages` branch, and they are the same as what we have on the `develop` branch. 
https://github.com/com-pas/compas-sct/blob/develop/docs/QUICK_START.md?plain=1#L48
```java
var scl = SclService.initScl(Optional.empty(), "1.0", "1.0");
```

File: docs/examples/example-app/src/main/java/org/lfenergy/compas/sct/examples/SctAppExample.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import org.lfenergy.compas.sct.commons.scl.SclRootAdapter;
+import org.lfenergy.compas.sct.commons.scl.SclService;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import java.util.Optional;
+import java.util.UUID;
+
+public class SctAppExample {
```
[the remark this answers is not in the document]
Comment (samirromdhani) [reply docs/examples/example-app/src/main/java/org/lfenergy/compas/sct/examples/SctAppExample.java:17]: These were just reported in the `gh-pages` branch and are identical to what is on the `develop` branch.

File: docs/examples/example-app/src/test/java/org/lfenergy/compas/sct/examples/SctAppExampleTest.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import javax.xml.bind.JAXBException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+public class SctAppExampleTest {
```
[the remark this answers is not in the document]
Comment (samirromdhani) [reply docs/examples/example-app/src/test/java/org/lfenergy/compas/sct/examples/SctAppExampleTest.java:19]: These were just reported in the `gh-pages` branch and are identical to what is on the `develop` branch.

File: docs/SCD_GENERATION.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  - AND having DA@fc="ST" OR "MX"
+- Then group the lines by ExtRef@pServT and @iedName and @ldInst and FlowKind property (and DA@fc for Report only)
+- Each group of lines is used to create a new DataSet and Control Block
+- Then for each ExtRef signal, create and populate ExtRef @srcXXX attributes
+
+### Modeling of functional specifications
+#### Sequence diagram
+The sequence diagram below summarize the process
+
+![img_2.png](images/SequenceDiagram-CompasSCT.png)
```
[the remark this answers is not in the document]
Comment (samirromdhani) [reply docs/SCD_GENERATION.md:58]: I just reported these in `gh-pages` branch, and they are the same as what we have on the `develop` branch.

[the remark this answers is not in the document]
Comment (AliouDIAITE) [reply docs/SCD_GENERATION.md:58]: ok issue #238 is create for this purpose

File: docs/examples/example-app/src/main/java/org/lfenergy/compas/sct/examples/SctAppExample.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (AliouDIAITE) [reply docs/examples/example-app/src/main/java/org/lfenergy/compas/sct/examples/SctAppExample.java:17]: ok issue #238 is create for this purpose

[the remark this answers is not in the document]
Comment (AliouDIAITE) [reply docs/examples/example-app/src/main/java/org/lfenergy/compas/sct/examples/SctAppExample.java:17]: ok issue #238 is create for this purpose

[the remark this answers is not in the document]
Comment (AliouDIAITE) [reply docs/examples/example-app/src/main/java/org/lfenergy/compas/sct/examples/SctAppExample.java:17]: ok issue #238 is create for this purpose

[the remark this answers is not in the document]
Comment (AliouDIAITE) [reply docs/examples/example-app/src/main/java/org/lfenergy/compas/sct/examples/SctAppExample.java:17]: ok issue #238 is create for this purpose.

File: docs/QUICK_START.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (AliouDIAITE) [reply docs/QUICK_START.md:48]: ok issue #238 is create for this purpose.

File: docs/examples/example-app/src/test/java/org/lfenergy/compas/sct/examples/SctAppExampleTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (AliouDIAITE) [reply docs/examples/example-app/src/test/java/org/lfenergy/compas/sct/examples/SctAppExampleTest.java:19]: ok issue #238 is create for this purpose.

[the remark this answers is not in the document]
Comment (AliouDIAITE) [reply docs/examples/example-app/src/test/java/org/lfenergy/compas/sct/examples/SctAppExampleTest.java:19]: ok issue #238 is create for this purpose.

[the remark this answers is not in the document]
Comment (AliouDIAITE) [reply docs/examples/example-app/src/test/java/org/lfenergy/compas/sct/examples/SctAppExampleTest.java:19]: ok issue #238 is create for this purpose.

**Pull request merged**